components/suggest/src/suggest.udl

```diff
@@ -38,7 +38,8 @@ interface Suggestion {
         string iab_category,
         string impression_url,
         string click_url,
-        string raw_click_url
+        string raw_click_url,
+        f64 score
```

Comment:
This will be a backward-incompatible change for iOS and Android—we'll need to add the field [here](https://github.com/mozilla-mobile/firefox-ios/blob/889120b984079af73e075b47dae918f4cba924da/firefox-ios/Storage/Rust/RustFirefoxSuggestion.swift#L42) for iOS, and the [tests](https://github.com/mozilla-mobile/firefox-android/blob/51f88bd6180d80fee4cc69d56607269455cf831c/android-components/components/feature/fxsuggest/src/test/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProviderTest.kt#L82-L101) on Android.

There's a bit of a [manual process](https://mozilla.github.io/application-services/book/howtos/breaking-changes.html#merging) for handling those:

1. Put up PRs for iOS and Android.
2. Get the iOS and Android PRs approved, but not landed yet.
3. Land this Application Services PR.
4. Wait for the automated PRs on Android and iOS that bump their Application Services versions to fail.
5. Merge the PRs from (1) into the version bump branches from (4).

Reply:
oh lol, sounds like im going on cross platform adventure 😂 , will do!

Reply:
TIL! Gotta say that process looks quite involved to a newbie like me. Out of curiosity, what happens if we land it w/o fixing breakages in downstream? Will we get backouts like what Desktop does :D?

Reply:
Yeah, it's pretty manual!

For Android and iOS, there are bots that automatically bump the Application Services dependencies. Here's an example for iOS: mozilla-mobile/firefox-ios#18117 and Android: mozilla-mobile/firefox-android#5130. On Android, if the tests pass, the PR is merged automatically; on iOS, someone from that team needs to manually trigger CI, approve, and merge.

On Android, if the PR fails on CI, the Release Owner usually pings `#application-services-eng` on Slack, at which point the author of the breaking change merges their PR _into the branch for the version bump PR_. Here's an example of that: mozilla-mobile/firefox-android#4088. That PR folded in my fix for a breaking change in Suggest, and a fix from James for a breaking change in Nimbus. Once that combined branch was green, it was merged using "squash and merge".

On iOS, if CI fails, `@firefox-ios-alerts-bitrise` will notify `#firefox-ios`—and someone from the team might tag you or someone from the Sync team. It's the same deal there—the author of the breaking change merges their PR into the branch with the version bump, and then someone from the iOS team re-triggers CI, and squash-merges once it's all green. Here's an example: mozilla-mobile/firefox-ios#16840

There are no backouts—Android and iOS will stay on the old Application Services versions until the breakage is resolved 😅 We try to avoid backward-incompatible changes (but sometimes they can't be helped, like here), and resolve them pretty quickly. Breakages don't prevent iOS and Android from building, but they do prevent them from getting more A-S version bumps.

(On Desktop, there's no automated vendoring at all 😢)

Reply:
Hey Lina, thank you so much for walking us through with examples! I have a better grasp on how things interact behind the scenes!

While we might be able to find those details in the [A-S doc](zilla.github.io/application-services/book/index.html), I just feel it's much easier to get up to speed with guided walk-through like this. This should be part of the "Hack A-S: Episode II" :)